### PR TITLE
patching logging

### PR DESCRIPTION
**Pull request by vktg**
Redmine Issue: https://redmine.pfsense.org/issues/9742
 Ready for review

Adds ability to log patch id to system log

### CODE REVIEW

**Comment by vktg** [comment]
+ code cleanup
+ description field is mandatory now
+ logging of **Auto Apply**

File: sysutils/pfSense-pkg-System_Patches/files/usr/local/www/system_patches.php [hunk 1]
```diff
@@ -48,28 +48,38 @@
...
-	$savemsg .= " (<a href=\"system_patches.php?id={$_POST['id']}&amp;fulltest=revert\" usepost>" . gettext("detail") . "</a>)";
-}
-if (($_POST['fulltest']) && ($a_patches[$_POST['id']])) {
-	if ($_POST['fulltest'] == "apply") {
-		$fulldetail = patch_test_apply($a_patches[$_POST['id']], true);
-	} elseif ($_POST['fulltest'] == "revert") {
-		$fulldetail = patch_test_revert($a_patches[$_POST['id']], true);
+if ($a_patches[$_POST['id']]) {
+	$savemsg = gettext("Patch ");
+	$descr = $a_patches[$_POST['id']]['descr'];
```
Comment (jim-p) [inline sysutils/pfSense-pkg-System_Patches/files/usr/local/www/system_patches.php]: What about previously saved patches that have no description?

This should probably log the patch `uniqid` and then also print the description, if present. The patch ID is shown on the page for every patch already.

File: sysutils/pfSense-pkg-System_Patches/files/usr/local/www/system_patches.php [hunk 2]
```diff
@@ -48,28 +48,38 @@
...
+	switch ($_POST['act']) {
+	case 'fetch':
```
Comment (jim-p) [inline sysutils/pfSense-pkg-System_Patches/files/usr/local/www/system_patches.php]: Everything in the `switch` needs indented one more tab

File: sysutils/pfSense-pkg-System_Patches/files/usr/local/www/system_patches_edit.php [hunk 3]
```diff
@@ -71,8 +71,8 @@
 
 	/* input validation */
 	if (empty($_POST['location'])) {
-		$reqdfields = explode(" ", "patch");
-		$reqdfieldsn = array(gettext("Patch Contents"));
+		$reqdfields = explode(" ", "descr patch");
```
Comment (jim-p) [inline sysutils/pfSense-pkg-System_Patches/files/usr/local/www/system_patches_edit.php]: I don't think description should be required. It wasn't in the past, and there are likely many patch entries already present without it. There is no compelling reason to force it.

**Comment by vktg** [comment]
+ tabs to switch statement
+ mandatory description removed
+ uniqid in log, descr if present

File: sysutils/pfSense-pkg-System_Patches/files/usr/local/pkg/patches.inc [hunk 4]
```diff
@@ -227,4 +229,19 @@ function patch_remove_shellcmd() {
...
 }
 
+function patchlog($msg) {
+        syslog(LOG_WARNING, gettext("System Patches: {$msg}"));
+}
+
+function patch_descr($id) {
+	global $config;
+	$a_patches = &$config['installedpackages']['patches']['item'];
+	$descr = gettext(" (ID: {$a_patches[$id]['uniqid']}");
```
Comment (jim-p) [inline sysutils/pfSense-pkg-System_Patches/files/usr/local/pkg/patches.inc]: The only part of this string that should be run through `gettext()` is `ID`. The rest is formatting and variables.

File: sysutils/pfSense-pkg-System_Patches/files/usr/local/pkg/patches.inc [hunk 5]
```diff
@@ -227,4 +229,19 @@ function patch_remove_shellcmd() {
...
+	if (isset($a_patches[$id]['descr'])) {
+		$descr .= gettext(", DESCR: {$a_patches[$id]['descr']}");
```
Comment (jim-p) [inline sysutils/pfSense-pkg-System_Patches/files/usr/local/pkg/patches.inc]: The only part of this string that should be run through `gettext()` is `DESCR`. The rest is formatting and variables. And that label is probably not necessary.

File: sysutils/pfSense-pkg-System_Patches/files/usr/local/pkg/patches.inc (hunk 4 above)
Comment (vktg) [reply sysutils/pfSense-pkg-System_Patches/files/usr/local/pkg/patches.inc]: gettext() fixed

File: sysutils/pfSense-pkg-System_Patches/files/usr/local/pkg/patches.inc (hunk 5 above)
Comment (vktg) [reply sysutils/pfSense-pkg-System_Patches/files/usr/local/pkg/patches.inc]: gettext() fixed

**Comment by vktg** [comment]
done

PORTREVISION +1

**Comment by vktg** [comment]
$descr = " (" . gettext("ID") . ": {$a_patches[$id]['uniqid']}"**)**;
                                                                                             ^

**Comment by vktg** [comment]
with fixes:
https://github.com/pfsense/FreeBSD-ports/pull/692

**Pull request merged**